modules/rest_api/controllers/services/rest.php

```diff
@@ -1832,19 +1856,88 @@ private function esGetSpecialFieldBlank(array $doc) {
    * @return string
    *   Combined comment string.
    */
-  private function esGetSpecialFieldSampleOccurrenceComment(array $doc) {
+  private function esGetSpecialFieldSampleOccurrenceComment(array $doc, array $params) {
     $oComment = isset($doc['occurrence']['occurrence_remarks']) ? $doc['occurrence']['occurrence_remarks'] : '';
     $sComment = isset($doc['event']['event_remarks']) ? $doc['event']['event_remarks'] : '';
+    if (!empty($params) && in_array("notab", $params)) {
+      $oComment = str_replace("\t", ' ', $oComment);
+      $sComment = str_replace("\t", ' ', $sComment);
+    }
+    if (!empty($params) && in_array("nonewline", $params)) {
+      $oComment = str_replace(array("\r\n", "\n", "\r"), ' ', $oComment);
+      $sComment = str_replace(array("\r\n", "\n", "\r"), ' ', $sComment);
+    }
     if ($oComment !== '' && $sComment !== '') {
-      return "Record comment: $oComment Sample comment: $sComment";
+      $comment = "$oComment $sComment";
     }
     elseif ($oComment !== '') {
-      return "Record comment: $oComment";
+      $comment =  $oComment;
     }
     elseif ($sComment !== '') {
-      return "Sample comment: $sComment";
+      $comment = $sComment;
     }
     else {
+      $comment = '';
+    }
+    if (!empty($params) && in_array("addref", $params)) {
+      //$ref = $doc['_id'];
+      //$ref = $doc['id'];
+      //$ref = $doc['occurrence']['id'];
```

Comment:
@johnvanbreda - each of these leads to an error. Any ideas what I am doing wrong here?

Reply:
`$ref = $doc['id']` should be correct. If you do this and combine with correct field request as in the earlier comment, what error do you get? Can you put a debug line in (kohana::log) so you can see the exact contents of $doc & $doc['id']?

Reply:
@johnvanbreda - I think I was fooled by caching or something since the `$ref = $doc['id']` method does indeed work.